docs/apis/calcite.md

```diff
@@ -0,0 +1,235 @@
+
+
+### Using User Defined functions ( UDF ) in SQL query
+
+We can use our own scalar UDF, implemented in Java, in a SQL statement for data manipulation. But before the usage, we need to register the function with execution environment by using registerFunction API.
+
+```java
+sqlEnv = sqlEnv.registerFunction("APEXCONCAT", PureStyleSQLApplication.class, "apex_concat_str");
+```
+
+In above code, **_registerFunction_** takes UDF name to be used in SQL, JAVA class which implements static method and name of the same method as parameters. 
+Used static method "apex_concat_str" takes two String objects as input parameters as a requirement from SQL query.
+```java    
+public static String apex_concat_str(String s1, String s2)
+{
+  return s1 + s2;
+}
+```
+
+The scalar UDF "APEXCONCAT" can be used in SQL as described below whereas FLOOR & SUBSTRING are standard SQL scalar functions supported by Apache Calcite.
+```sql
+INSERT INTO SALES 
+       SELECT STREAM ROWTIME, FLOOR(ROWTIME TO DAY), APEXCONCAT('OILPAINT', SUBSTRING(PRODUCT, 6, 7)) 
+       FROM ORDERS 
+       WHERE ID > 3 AND PRODUCT LIKE 'paint%'
+```
+To read about all functions and operators supported by Apache Calcite, click [here](https://calcite.apache.org/docs/reference.html#operators-and-functions).
+
+### Executing SQL Query 
+Finally to execute the query we need to use **_executeSQL_** API with DAG and SQL statement as parameters.
+
+```java
+// Converting SQL statement to DAG 
+String sql = "INSERT INTO SALES 
+                SELECT STREAM ROWTIME, FLOOR(ROWTIME TO DAY), APEXCONCAT('OILPAINT', SUBSTRING(PRODUCT, 6, 7)) 
+                FROM ORDERS 
+                WHERE ID > 3 AND PRODUCT LIKE 'paint%'";
+sqlEnv.executeSQL(dag, sql);
+```
+When executeSQL API is called, specified SQL query go through various phases like conversion to relational algebra, optimization and planning in Calcite to generate Relation Expression Tree. 
+Generated Relation Expression Tree is consumed by Apex SQL and converted to DAG using operators available in Apache Malhar. Particularly in this example, ORDERS table definition will be converted to "KafkaInputOperator+CSVParser" and "SALES" table definition will be converted to "CSVFormatter+FileOutputFormatter".
+ _"WHERE"_ clause is used in this SQL query, which defines filter operation and converted to _FilterTransformOperator_. Based on condition for projection in relational algebra, projection operator is added to DAG which filters out unwanted columns as well applies user defined functions on data. This _projection operator_ is implemented using _FilterTransformOperator_ in SQL on Apex. DAG created for this application will look like as shown below.
```

Comment:
Start a new paragraph for this line.
The fact that there are 2 FilterTransformOperators needs to be explained better, perhaps something like this:

"_"WHERE"_ clause ... on Apex." => A _WHERE_ clause is used in this query; it defines the desired filter for rows and is converted to a _FilterTransformOperator_ in the DAG. Similarly, the projection defining desired columns is converted into another instance of the _FilterTransformOperator_.

DAG created ... shown below. => The DAG created for this application will look like this: